File: streamline/src/main/java/me/snowmii/streamline/ImageBinding.java
package me.snowmii.streamline;

/**
 * One engine image: raw handles and format. Subresource range is absent: Minecraft Vulkan
 * images are single-level, single-layer 2D, and native derives range and aspect from the
 * binding's role.
 */
public record ImageBinding(
	long view,
	long image,
	int format
) {}

File: src/main/kotlin/me/snowmii/dlss/bridge/ImageBinding.kt
package me.snowmii.dlss.bridge

/**
 * One engine image handed to the native ABI: its raw handles and format.
 *
 * The subresource range is deliberately absent. Every image Minecraft's Vulkan backend creates
 * is a single-level, single-layer 2D image, and the native side derives the range - and the
 * aspect, from the binding's role - rather than having it carried. The ABI used to carry all
 * five range fields per image and every producer sent the same values.
 */
data class ImageBinding(
	val view: Long,
	val image: Long,
	val format: Int,
)
